nakadi-producer-spring-boot-starter/src/main/java/org/zalando/nakadiproducer/NakadiProducerAutoConfiguration.java

```diff
@@ -73,6 +77,12 @@ public StupsTokenComponent accessTokenProvider(
         }
     }
 
+    @Bean
+    @ConditionalOnMissingBean
+    RequestFactory requestFactory(@Value("${nakadi-producer.encoding:GZIP}") ContentEncoding encoding){
+        return new SimpleRequestFactory(encoding);
+    }
```

Comment:
We only use the RequestFactory in line 62, so it can be inside the same conditional configuration, I think.
(No need to pollute the bean space with a RequestFactory which we are not using ourselves.)

```suggestion

        @Bean
        @ConditionalOnMissingBean
        RequestFactory requestFactory(@Value("${nakadi-producer.encoding:GZIP}") ContentEncoding encoding){
            return new SimpleRequestFactory(encoding);
        }
    }
```

Reply:
>   NakadiProducerAutoConfiguration.FahrscheinWithTokensNakadiClientConfiguration:
>      Did not match:
>
>   - @ConditionalOnMissingBean (types: org.zalando.nakadiproducer.transmission.NakadiPublishingClient,org.zalando.fahrschein.NakadiClient; SearchStrategy: all) found beans of type 'org.zalando.nakadiproducer.transmission.NakadiPublishingClient' nakadiClient (OnBeanCondition)
```

Reply:
Hmm, I'm not quite happy with this, but I don't really have a solution. I guess we can also fix this later.

Reply:
I think I found a solution, let me push the changes.